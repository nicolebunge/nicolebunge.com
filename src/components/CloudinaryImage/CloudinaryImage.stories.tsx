import CloudinaryImage from '.';

export default {
  component: CloudinaryImage,
};

export const Default = {
  args: {
    alt: 'Image alt text',
    src: 'path/to/image.jpg',
    width: 400,
    height: 300,
  },
};
