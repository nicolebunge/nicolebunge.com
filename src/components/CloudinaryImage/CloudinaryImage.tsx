/* eslint-disable jsx-a11y/alt-text */

import React from 'react';
import { buildQuery } from '../../utils/cloudinary';

type PublicIds =
  | 'nicolebunge.com/gallery/alice.jpg'
  | 'nicolebunge.com/gallery/berliner.jpg'
  | 'nicolebunge.com/gallery/das-tote-maedchen.jpg'
  | 'nicolebunge.com/gallery/goth.jpg'
  | 'nicolebunge.com/gallery/image-01-large.jpg'
  | 'nicolebunge.com/gallery/image-01.jpg'
  | 'nicolebunge.com/gallery/image-02-large.jpg'
  | 'nicolebunge.com/gallery/image-02.jpg'
  | 'nicolebunge.com/gallery/image-03-large.jpg'
  | 'nicolebunge.com/gallery/image-03.jpg'
  | 'nicolebunge.com/gallery/kompagnie-1.jpg'
  | 'nicolebunge.com/gallery/kompagnie-2.jpg'
  | 'nicolebunge.com/gallery/portrait/10_kqpbxw.jpg'
  | 'nicolebunge.com/tanz-rumba.jpg'
  | 'nicolebunge.com/home/theaterpaedagogik.jpg'
  | 'nicolebunge.com/nicole-bunge.svg';

interface CloudinaryImageProps {
  alt: string;
  height: number;
  src: PublicIds;
  width: number;
}

function CloudinaryImage(props: CloudinaryImageProps): JSX.Element {
  const { height, src, width, ...otherProps } = props;
  const query = buildQuery({ height, width });

  return (
    <img
      height={height}
      loading="lazy"
      src={`https://res.cloudinary.com/nicolebunge/image/upload/${query}/v1627923793/${src}`}
      width={width}
      {...otherProps}
    />
  );
}

export type { CloudinaryImageProps };
export default CloudinaryImage;
