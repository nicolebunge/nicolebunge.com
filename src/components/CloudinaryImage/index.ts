import CloudinaryImage from './CloudinaryImage';

export * from './CloudinaryImage';
export default CloudinaryImage;
