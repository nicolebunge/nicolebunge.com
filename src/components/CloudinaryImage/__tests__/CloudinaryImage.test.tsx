import { composeStories } from '@storybook/testing-react';
import { render } from '@testing-library/react';
import React from 'react';
import * as stories from '../CloudinaryImage.stories';

const { Default } = composeStories(stories);

describe('Cloudinary image component', () => {
  it('renders correctly', () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
