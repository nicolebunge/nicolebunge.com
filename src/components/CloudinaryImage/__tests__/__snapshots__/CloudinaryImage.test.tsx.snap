// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Cloudinary image component > renders correctly 1`] = `
<img
  alt="Image alt text"
  height="300"
  loading="lazy"
  src="https://res.cloudinary.com/nicolebunge/image/upload/c_fill,f_auto,h_300,q_auto,w_400/v1627923793/path/to/image.jpg"
  width="400"
/>
`;
